File: TwoWayDatabinding/app/src/main/java/com/enpassio/twowaydatabinding/ui/MainActivity.java
package com.enpassio.twowaydatabinding.ui;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;

import com.enpassio.twowaydatabinding.R;
import com.enpassio.twowaydatabinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        setSupportActionBar(binding.toolbar);

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.main_container, new ToyListFragment())
                    .commit();
        }
    }

    @Override
    public void onBackPressed() {
        Fragment currentFrag = getSupportFragmentManager().findFragmentById(R.id.main_container);
        if (currentFrag instanceof AddToyFragment) {
            ((AddToyFragment)currentFrag).onBackClicked();
        } else {
            super.onBackPressed();
        }
    }

}

File: TwoWayDatabindingKotlin/app/src/main/java/com/enpassion/twowaydatabindingkotlin/ui/MainActivity.kt
package com.enpassion.twowaydatabindingkotlin.ui

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.enpassion.twowaydatabindingkotlin.R
import com.enpassion.twowaydatabindingkotlin.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        setSupportActionBar(binding.toolbar)

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                .add(R.id.main_container, ToyListFragment())
                .commit()
        }
    }

    override fun onBackPressed() {
        /*//If back is clicked when AddToyFragment is on the screen,
        check whether there are unsaved changes*/
       val currentFrag = supportFragmentManager.findFragmentById(R.id.main_container)
        if(currentFrag is AddToyFragment){
            currentFrag.onBackClicked()
        } else {
            super.onBackPressed()
        }
    }
}
